src/core/swap.c

```diff
@@ -290,7 +290,7 @@ static int swap_load(Unit *u) {
 
         if (u->load_state == UNIT_LOADED) {
 
-                if (UNIT(s)->fragment_path)
+                if (UNIT(s)->fragment_path && UNIT(s)->source_path)
```

Comment:
nah, this is not an acceptable patch, when something has a fragment, that's what counts.